recipes/aws-kvs-pic/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class awskvspicConan(ConanFile):
+    name = "aws-kvs-pic"
+    license = "Apache-2.0"
+    homepage = "https://github.com/awslabs/amazon-kinesis-video-streams-pic"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("Platform Independent Code for Amazon Kinesis Video Streams")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {'shared': False, "fPIC": True}
+    generators = "cmake"
+    topics = ("conan", "aws", "kvs", "stream")
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_DEPENDENCIES"] = False
+            self._cmake.configure()
+        return self._cmake
+
+    def validate(self):
+        if (self.settings.os != "Linux" and self.options.shared):
+            raise ConanInvalidConfiguration("This library can only be built shared on Linux")
```

Comment:
Even not on Macos?

Reply:
Perhaps also also include FreeBSD?

Reply:
Nope. Take a look at the previous builds of this PR. 
Until like a week ago this lib was STATIC only, it was hardcoded. Their CCI only tests static.
It is not surprising to me that shared builds are broken.

Reply:
https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/6716/7-production/aws-kvs-pic/cci.20210705//summary.json

Reply:
I looked a bit at the cmake script and I think the `kvspicState` library is not linked correctly. It needs to link to kvspicUtils as well.
```patch
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,7 +149,7 @@ add_library(kvspicState ${LIBTYPE} ${PIC_STATE_SOURCE_FILES})
 if(BUILD_SHARED_LIBS)
   set_target_properties(kvspicState PROPERTIES VERSION 0.0.0 SOVERSION 0)
 endif()
-target_link_libraries(kvspicState)
+target_link_libraries(kvspicState kvspicUtils)
 
 configure_file(
   "${CMAKE_CURRENT_SOURCE_DIR}/libkvspicUtils.pc.cmake"
```

`kvspicClient` has a similar problem, but I could not figure out from the logs to what libraries it has to link to.

Reply:
There are more than one line like that links to nothing. Also the kvspic lib is a superset of the other 3, take a look a its sources. I opened an issue on their GitHub asking if that was intentional and they say yes.
This library is weird.

Reply:
Indeed. They also have a cmake script in each `src/xxx/src` subdirectory. But don't include it.